cds_migrator_kit/circulation/items/api.py

```diff
@@ -0,0 +1,210 @@
+        """Return the new record circulation restriction."""
+        legacy_restriction = item['loan_period']
+        if legacy_restriction == 'Reference':
+            new_restriction = self.RESTRICTIONS[0]
+        elif legacy_restriction in ['4 weeks', '1 week']:
+            new_restriction = None
+        else:
+            raise ValueError
+        return new_restriction
+
+    def _clean_date(self, legacy_date):
+        """Return the new record date format."""
+        # old: 2016-01-29T17:28:17
+        # new: "2018-05-16T12:34:28.233187+00:00"
+        d = datetime.strptime(legacy_date, '%Y-%m-%dT%H:%M:%S')
+        return d.isoformat()
+
+    def migrate(self):
+        """Return items."""
+        records = []
+        for i, item in enumerate(self.items):
+
+            # library id
+            if item['id_crcLIBRARY'] not in self.internal_locations:
+                print(
+                    "Item with `barcode = {barcode}` not imported because "
+                    "`library_id={libid}` not found in imported "
+                    "libraries".format(
+                        barcode=item['barcode'], libid=item['id_crcLIBRARY']))
+                continue
+            ilocid = self.internal_locations[item['id_crcLIBRARY']]
+
+            # status
+            try:
+                status = self._transform_status(item)
+            except ValueError:
+                print(
+                    "Item with `barcode = {barcode}` not imported because "
+                    "`status={status}` not recognized".format(
+                        barcode=item['barcode'], status=item['status']))
+                continue
+
+            # circulation_restriction
+            try:
+                circulation_restriction = self._transform_loan_period(item)
+            except ValueError:
+                print(
+                    "Item with `barcode = {barcode}` not imported because "
+                    "`loan_period={loan_period}` not recognized".format(
+                        barcode=item['barcode'],
+                        loan_period=item['loan_period']))
+                continue
+
+            try:
+                created = self._clean_date(item['creation_date'])
+            except ValueError:
+                created = None
+
+            try:
+                updated = self._clean_date(item['modification_date'])
+            except ValueError:
+                updated = None
+
+            record = {
+                "itemid": i,
+                "docid": "to be set",
+                "ilocid": ilocid,
+                "legacy_bibrec_id": item['id_bibrec'],
+                "legacy_library_id": item['id_crcLIBRARY'],
+                "barcode": item['barcode'],
+                "shelf": item['location'],
+                "description": item['description'],
+                "status": status,
+                "circulation_restriction": circulation_restriction,
+                "medium": "",
```

Comment:
just a question, because maybe I've missed something, why the medium is empty string always?

Reply:
Maybe we can extract this info in a second moment from the Document, but for now, it is not set.